shared/constants/network.ts

```diff
@@ -0,0 +1,579 @@
+  [CHAIN_IDS.LOCALHOST]: NETWORK_IDS.LOCALHOST,
+  [CHAIN_IDS.RINKEBY]: NETWORK_IDS.RINKEBY,
+  [CHAIN_IDS.ROPSTEN]: NETWORK_IDS.ROPSTEN,
+} as const;
+
+export const NATIVE_CURRENCY_TOKEN_IMAGE_MAP = {
+  [CURRENCY_SYMBOLS.ETH]: ETH_TOKEN_IMAGE_URL,
+  [CURRENCY_SYMBOLS.TEST_ETH]: TEST_ETH_TOKEN_IMAGE_URL,
+  [CURRENCY_SYMBOLS.BNB]: BNB_TOKEN_IMAGE_URL,
+  [CURRENCY_SYMBOLS.MATIC]: MATIC_TOKEN_IMAGE_URL,
+  [CURRENCY_SYMBOLS.AVALANCHE]: AVAX_TOKEN_IMAGE_URL,
+} as const;
+
+export const INFURA_BLOCKED_KEY = 'countryBlocked';
+
+/**
+ * Hardforks are points in the chain where logic is changed significantly
+ * enough where there is a fork and the new fork becomes the active chain.
+ * These constants are presented in chronological order starting with BERLIN
+ * because when we first needed to track the hardfork we had launched support
+ * for EIP-2718 (where transactions can have types and different shapes) and
+ * EIP-2930 (optional access lists), which were included in BERLIN.
+ *
+ * BERLIN - forked at block number 12,244,000, included typed transactions and
+ * optional access lists
+ * LONDON - future, upcoming fork that introduces the baseFeePerGas, an amount
+ * of the ETH transaction fees that will be burned instead of given to the
+ * miner. This change necessitated the third type of transaction envelope to
+ * specify maxFeePerGas and maxPriorityFeePerGas moving the fee bidding system
+ * to a second price auction model.
+ */
+export const HARDFORKS = {
+  BERLIN: 'berlin',
+  LONDON: 'london',
+} as const;
+
+export const CHAIN_ID_TO_GAS_LIMIT_BUFFER_MAP = {
+  [CHAIN_IDS.OPTIMISM]: 1,
+  [CHAIN_IDS.OPTIMISM_TESTNET]: 1,
+};
+
+/**
+ * Ethereum JSON-RPC methods that are known to exist but that we intentionally
+ * do not support.
+ */
+export const UNSUPPORTED_RPC_METHODS = new Set([
+  // This is implemented later in our middleware stack – specifically, in
+  // eth-json-rpc-middleware – but our UI does not support it.
+  'eth_signTransaction' as const,
+]);
+
+export const IPFS_DEFAULT_GATEWAY_URL = 'dweb.link';
+
+// The first item in transakCurrencies must be the
+// default crypto currency for the network
+const BUYABLE_CHAIN_ETHEREUM_NETWORK_NAME = 'ethereum';
+
+type MoonPayNetworkAbbreviations = 'bsc' | 'cchain' | 'polygon';
+
+type MoonPayChainSettings = {
+  defaultCurrencyCode:
+    | LowercaseCurrencySymbol
+    | `${LowercaseCurrencySymbol}_${MoonPayNetworkAbbreviations}`;
+  showOnlyCurrencies:
+    | `${LowercaseCurrencySymbol}`
+    | `${LowercaseCurrencySymbol},${LowercaseCurrencySymbol}`
+    | `${LowercaseCurrencySymbol},${LowercaseCurrencySymbol},${LowercaseCurrencySymbol}`
+    | `${LowercaseCurrencySymbol},${LowercaseCurrencySymbol},${LowercaseCurrencySymbol},${LowercaseCurrencySymbol}`
+    | `${LowercaseCurrencySymbol}_${MoonPayNetworkAbbreviations}`
+    | `${LowercaseCurrencySymbol}_${MoonPayNetworkAbbreviations},${LowercaseCurrencySymbol}_${MoonPayNetworkAbbreviations}`;
```

Comment:
This type matches either 1,2 or 3 currency symbols in lower case format separated by ','. It also will match either 1 or 2 currency symbols in lowercase format, concatenated with a network abbreviation and '_' separated by ','.

Reply:
> +export const TEST_NETWORK_TICKER_MAP = {

+  [NETWORK_TYPES.ROPSTEN]: `${

+    capitalize(NETWORK_TYPES.ROPSTEN) as Capitalize<

+      typeof NETWORK_TYPES.ROPSTEN

+    >

+  }${CURRENCY_SYMBOLS.ETH}`,

+  [NETWORK_TYPES.RINKEBY]: `${

+    capitalize(NETWORK_TYPES.RINKEBY) as Capitalize<

+      typeof NETWORK_TYPES.RINKEBY

+    >

+  }${CURRENCY_SYMBOLS.ETH}`,

+  [NETWORK_TYPES.KOVAN]: `${

+    capitalize(NETWORK_TYPES.KOVAN) as Capitalize<typeof NETWORK_TYPES.KOVAN>

+  }${CURRENCY_SYMBOLS.ETH}`,

+  [NETWORK_TYPES.GOERLI]: `${

+    capitalize(NETWORK_TYPES.GOERLI) as Capitalize<typeof NETWORK_TYPES.GOERLI>

+  }${CURRENCY_SYMBOLS.ETH}`,

+} as const;

Reply:
Nice! I love it.